tests/thread_sleep/tests/01-run.py

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env python3
+
+# Copyright (C) 2022 Benjamin Valentin <benjamin.valentin@ml-pa.com>
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+
+import sys
+from testrunner import run
+
+
+def testfunc(child):
+    child.expect('first thread')
+    child.expect('second thread')
+    child.expect('> wake sleeping threads')
+    child.expect('first thread')
+    child.expect('second thread')
+    child.expect('> put threads to sleep again')
```

Comment:
As you are not expecting any regexes, it is preferred to use `expect_exact`.

```suggestion
    child.expect_exact("first thread")
    child.expect_exact("second thread")
    child.expect_exact("> wake sleeping threads")
    child.expect_exact("first thread")
    child.expect_exact("second thread")
    child.expect_exact("> put threads to sleep again")
```

Also please use double quotes, as this is preferred by python linters such as `black`.